imports/client/components/GuessQueuePage.tsx

```diff
@@ -138,17 +145,91 @@ type GuessQueuePageProps = GuessQueuePageParams & {
 };
 
 class GuessQueuePage extends React.Component<GuessQueuePageProps> {
+  searchBarRef?: HTMLInputElement
+
+  onSearchStringChange = (e: React.FormEvent<FormControl>) => {
+    this.setSearchString((e as unknown as React.FormEvent<HTMLInputElement>).currentTarget.value);
+  };
+
+  getSearchString = (): string => {
+    return this.props.location.query.q || '';
+  };
+
+  setSearchString = (val: string) => {
+    const newQuery = val ? { q: val } : { q: undefined };
+    browserHistory.replace({
+      pathname: this.props.location.pathname,
+      query: { ...this.props.location.query, ...newQuery },
+    });
+  };
+
+  clearSearch = () => {
+    this.setSearchString('');
+  }
+
+  compileMatcher = (searchKeys: string[]): (g: GuessType) => boolean => {
+    return (guess) => {
+      const puzzle = this.props.puzzles[guess.puzzle];
+      const guessText = guess.guess.toLowerCase();
+
+      // Try to find a reason to be interested in this guess.
+      for (let i = 0; i < searchKeys.length; i++) {
+        const key = searchKeys[i].toLowerCase();
+        if (guessText.indexOf(key) !== -1 || puzzle.title.toLowerCase().indexOf(key) !== -1) {
```

Comment:
Is matching arbitrary substrings in the title the user-expected behavior? I think I’d expect it to be constrained to leading (or potentially leading or trailing) substrings of each word in the title, especially given the implicit disjunction. (For guesses, which frequently lack word breaks, arbitrary substrings make sense of course.)

You should do whatever you prefer, as it makes no difference and I don’t think we have a consistent style, but in my opinion using searchKeys.some() would be more consistent with the rest of the code (and more concise) than this for-loop.

Reply:
Grant, you have a knack for asking the right questions to make me think about a problem more clearly. :)

Thoughts:

* I do explicitly want the filter to be type-ahead style filtering, where adding additional constraints will never make the list of results get longer.  I like having the live-update on each keystroke, and making sure that the results have this property of stability is something that I value.
* On reflection, I'm not sure having disjunctive search results makes the most sense here, or anywhere?  The original implementation for filter-search in the puzzle list page is disjunctive, and I think that might have been because the control flow was easier for December 2015 Drew to hack together quickly and get working, and not because I thought about it real hard and concluded this was the best approach.
* Arbitrary substring matching is probably not quite how people usually use search.  I think I started out with the attitude that it was better to make it easier to get a match than not in case there's punctuation or something, but that might also have been that it was fewer function calls to just `indexOf()` the whole title rather than doing word-splitting on each puzzle title.  That said, I can't remember a time that the permissive mid-word-matching has been useful rather than surprising, so I'm willing to try more restricted matching here
* Prefix-matching is compatible with the constraints around incrementally shrinking the result set; postfix-matching is not.

I think I'll switch this to doing conjunction on multiple keys, prefix search on title words, and keep looking for any substring in answer phrases.